### PR TITLE
Allow `labelFormat` option to be a function

### LINKED ISSUE #1559: More options/values for labelFormat
**The problem**

For the moment I use `"labelFormat": "[dirname]-[filename]--[local]",` but that's not enough for me.

**Proposed solution**

I would like that `dirname` takes all the folders.

**Context**

I'm here:  `/src/pages/Home/Main/Content.js` with: 

```javascript
const Heading = styled.div``

class HomePageMainContent extends React.Component {}
```

I would like that the className generated for `Heading` is:

- either `xxx-HomePageMainContent--Heading` (`random-NameOfTheReactClass--NameOfTheLocal`)
- or `xxx-pages-Home-Main-Content--Heading` (`random-The-Whole-Path--NameOfTheLocal`) 

Thank you :)

### PR DESCRIPTION
**Pull request by Andarist**
closes #1559

### CODE REVIEW

File: packages/babel-plugin-emotion/src/utils/label.js [hunk 1]
```diff
@@ -9,11 +14,23 @@ const sanitizeLabelPart = (labelPart: string) =>
...
+  const sanitizedName = sanitizeLabelPart(identifierName)
+
+  if (!labelFormat) {
+    return sanitizedName
+  }
+
+  if (typeof labelFormat === 'function') {
+    return labelFormat({
+      name: sanitizedName,
+      filename
```
Comment (mitchellhamilton) [inline packages/babel-plugin-emotion/src/utils/label.js]: Could we call this `path` to distinguish it from the filename used in the string form of labelFormat?

Comment (Andarist) [reply packages/babel-plugin-emotion/src/utils/label.js]: Sure, changed.

File: packages/babel-plugin-emotion/src/utils/label.js [hunk 2]
```diff
@@ -1,6 +1,11 @@
 // @flow
 import nodePath from 'path'
 
+type LabelFormatOptions = {
+  name: string,
+  filename: string
```
Comment (mitchellhamilton) [inline packages/babel-plugin-emotion/src/utils/label.js]:
```suggestion
  path: string
```

Comment (Andarist) [reply packages/babel-plugin-emotion/src/utils/label.js]: 👍  just fixed it

**Pull request merged**